lua/lazyvim/plugins/extras/lang/fsharp.lua

```diff
@@ -0,0 +1,63 @@
+-- requires dotnet sdk  to be installed
+return {
+  recommended = function()
+    return LazyVim.extras.wants({
+      ft = { "fs", "fsx", "fsi" },
+      root = { "*.fsproj" },
+    })
+  end,
+  {
+    "nvim-treesitter/nvim-treesitter",
+    config = function()
+      local parser_config = require("nvim-treesitter.parsers").get_parser_configs()
+      parser_config.fsharp = {
+        install_info = {
+          url = "https://github.com/ionide/tree-sitter-fsharp",
+          branch = "main",
+          files = { "src/scanner.c", "src/parser.c" },
+        },
+        filetype = "fsharp",
+      }
+    end,
+    opts = { ensure_installed = { "fsharp" } },
+  },
+  {
+    "ionide/Ionide-vim",
+    ft = { "fs", "fsi", "fsx" },
+  },
+  {
+    "mason.nvim",
+    opts = { ensure_installed = {
+      "fantomas",
+    } },
+  },
+  {
+    "nvimtools/none-ls.nvim",
+    optional = true,
+    opts = function(_, opts)
+      local nls = require("null-ls")
+      opts.sources = vim.list_extend(opts.sources or {}, {
+        nls.builtins.formatting.fantomas,
+      })
+    end,
+  },
+  {
+    "stevearc/conform.nvim",
+    optional = true,
+    opts = {
+      formatters_by_ft = {
+        fs = { "fantomas" },
+        fsi = { "fantomas" },
+        fsx = { "fantomas" },
+      },
+    },
+  },
+  {
+    "neovim/nvim-lspconfig",
+    opts = {
+      servers = {
+        fsautocomplete = {},
+      },
+    },
+  },
```

Comment:
Should this be enabled with the `Ionide-vim`? From its [README](https://github.com/ionide/Ionide-vim?tab=readme-ov-file#for-neovim-05)
> If you are using [neovim/nvim-lspconfig](https://github.com/neovim/nvim-lspconfig), do not enable fsautocomplete. Ionide-vim automatically integrates itself into nvim-lspconfig and will register itself as a server.

Would it be better to hold up on this PR until more people who do f-sharp weigh in their opinions?

Reply:
My mistake, I didn't read that part in the readme, if you are right, this pr needs to be holded

Reply:
Do you actually program in f-sharp? Or did you only want to provide support in LazyVim for f-sharp because there was a request? This is actually a naive question as I don't know anything about you. But I believe that language Extras should actually be provided by people with experience in the specific language domain.

Reply:
I did csharp before and know a lil bit of fsharp , and have great idea of dotnet ecosystem so that's why I searched about these things and implemented them

Reply:
My personal opinion is to hold up on this PR since there are quite a few lacking areas that we aren't sure about. I'd prefer to wait for more people who do f-sharp to weigh in their opinions. But in the end it's up to the maintainer to decide.

Reply:
Good point.
Just updated the CONTRIBUTING doc.

@muneebusmani new languages really should only be added by people that use these languages already and have a lot of experience in configuring the language for Neovim.

You have to understand that when I merge a PR for a new plugin/language, then that also needs to be maintained. To lessen the burden on maintenance, the new lang should at least be configured by someone that knows the config well.

Thank you for the effort, but I'm going to close this for now.